core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -490,8 +491,9 @@ class MapUnit {
                     tile.terrainFeatures.remove(removedFeatureName)
                 }
             }
-            tile.improvementInProgress == "Road" -> tile.roadStatus = RoadStatus.Road
-            tile.improvementInProgress == "Railroad" -> tile.roadStatus = RoadStatus.Railroad
+            //todo: is it possible worldscreen-visible stats change by connecting stuff?
```

Comment:
Yes, connecting cities by railroad for example adds a whopping +25% to production, which affects construction times

Reply:
:+1: - of course, I've even seen the code lately. But not this PR, so a glaring yellow todo is just fine methinks

Reply:
No, I removed the comment as this is in next-turn code, not interactive? Not quite scope of this PR anyway.